storage/simulation/script.go

```diff
@@ -0,0 +1,310 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Bram Gruneir (bram+code@cockroachlabs.com)
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/cockroachdb/cockroach/util"
+)
+
+// Operation enumerates the possible scripted operations that can occur to a
+// cluster.
+type Operation int
+
+// These are the possible scripted operations.
+const (
+	OpSplitRange Operation = iota
+	OpAddNode
+	OpExit
+	// TODO(brma): add optoinal value to addnode to indicated size
+	// TODO(bram): consider many other operations here.
+	//	OpMergeRangeRandom
+	//	OpMergeRangeFirst
+	//  OpKillNodeRandom
+	//  OpKillNodeFirst
+	//  OpAddStoreRandom
+	//  OpAddStore
+	//	OpKillStoreRandom
+	//	OpKillStoreFirst
+)
+
+var operations = [...]string{
+	OpSplitRange: "splitrange",
+	OpAddNode:    "addnode",
+	OpExit:       "exit",
+}
+
+// String returns the string version of the Operation enumerable.
+func (operation Operation) String() string {
+	return operations[operation]
+}
+
+// operationFromString returns the operation corresponding with the passed in
+// string.
+func operationFromString(s string) (Operation, bool) {
+	for i, op := range operations {
+		if op == s {
+			return Operation(i), true
+		}
+	}
+	return 0, false
+}
+
+// OperationValue enumerates the possible values for a scripted operation.
+type OperationValue int
+
+// These are the possible operations values.
+const (
+	OpValSet OperationValue = iota
```

Comment:
Similar reply to the previous one.  Also, this matches the suggested style for enums.